satellite/nodestats/endpoint.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package nodestats
+
+import (
+	"context"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+)
+
+var (
+	// NodeStatsEndpointErr is endpoint error class
+	NodeStatsEndpointErr = errs.Class("node stats endpoint error")
+
+	mon = monkit.Package()
+)
+
+// Endpoint for querying node stats for the SNO
+type Endpoint struct {
+	log     *zap.Logger
+	overlay overlay.DB
+}
+
+// NewEndpoint creates new endpoint
+func NewEndpoint(log *zap.Logger, overlay overlay.DB) *Endpoint {
+	return &Endpoint{
+		log:     log,
+		overlay: overlay,
+	}
+}
+
+// UptimeCheck returns uptime checks ratio
+func (e *Endpoint) UptimeCheck(ctx context.Context, req *pb.UptimeRequest) (_ *pb.UptimeResponse, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	peer, err := identity.PeerIdentityFromContext(ctx)
+	if err != nil {
+		return nil, NodeStatsEndpointErr.Wrap(err)
+	}
+
+	node, err := e.overlay.Get(ctx, peer.ID)
+	if err != nil {
+		return nil, NodeStatsEndpointErr.Wrap(err)
+	}
+
+	return &pb.UptimeResponse{
+		Ratio: node.Reputation.UptimeRatio,
```

Comment:
We are no longer updating `UptimeRatio` and will likely remove it in the near future. If you still want to use it, I would recommend calculating it yourself in this function with `node.Reputation.UptimeSuccessCount / node.Reputation.UptimeCount`. Otherwise it will be inaccurate.

If you want to send the reputation value we are actually using to determine whether a node is good or bad, you can return `reputation = node.Reputation.UptimeReputationAlpha / (node.Reputation.UptimeReputationAlpha + node.Reputation.UptimeReputationBeta)`, which will be a number bounded between 0 and 1, where closer to 1 is a better reputation and closer to 0 is a worse reputation.

Reply:
thx for clarification!